### DISCUSSION TITLE
Phone Auth verification SMS contains ***.firebaseapp.com as the APP_NAME when using reCAPTCHA

### DISCUSSION THREAD

**Discussion by appsgenie** [invertase/react-native-firebase#7196] [Q&A]
Has anyone ran into this issue?
When the flow is able to avoid reCAPTCHA, the app store/play store name is included in the SMS, but when it goes to verify via reCAPTCHA, it show the ugly firebase name in the format of `project-name.firebaseapp.com`  :(

I made sure project settings in the Firebase Console are set correctly including the APP_NAME setting (for public facing name).

I think this might be tied to the fact that captcha is using an ugly firebase URL and not the custom domain I have configured?

**Comment by richard-rowdy-jr** [comment]
Looking for the same solution as well :eyes:

**Comment by balloman** [comment]
Did you find a solution?

**Comment by ansh** [comment]
I'm looking for a solution as well!

**Comment by jgadjetx** [comment]
Has anyone found a solution

**Comment by sayan-avantai** [comment]
As of now, sms template can't be changed. you can change only app name. but you can add your custom domain in **Authorized domains** in firebase settings.

**Comment by Prashamtogadiya** [comment]
@appsgenie @richard-rowdy-jr @sayan-avantai @jgadjetx @ansh @balloman 

Expected behavior.

When Play Integrity / APNs works, SMS uses the store app name. When it falls back to reCAPTCHA, %APP_NAME% becomes project-id.firebaseapp.com. Public-facing name in Firebase settings doesn’t fix that.

Fix: serve Auth on a custom domain.

1. Firebase Console → Hosting → deploy once if needed → Add custom domain (e.g. auth.yourdomain.com)
2. Add the DNS records (TXT + CNAME/A) Firebase shows in DNS provider 
3. Auth → Settings → Authorized domains → add that custom domain 
4. Point the app at it:
   - Web: authDomain: "auth.yourdomain.com"
   - Android: FirebaseAuth.getInstance().setCustomAuthDomain("auth.yourdomain.com")
   - iOS: Auth.auth().customAuthDomain = "auth.yourdomain.com"

After that, reCAPTCHA SMS / browser shows your domain instead of *.firebaseapp.com. You still can’t fully customize the SMS template.

**Reply by khayargoli** [reply]
how do I do it from my react-native app on android/ios?
